src/shared/clients/ec2Client.ts

```diff
@@ -62,19 +62,48 @@ export class Ec2Client {
         return response[0]
     }
 
-    /**
-     * Retrieve IAM role attached to given EC2 instance.
-     * @param instanceId target EC2 instance ID
-     * @returns IAM role associated with instance, or undefined if none exists.
-     */
-    public async getAttachedIamRole(instanceId: string): Promise<IamInstanceProfile | undefined> {
-        const client = await this.createSdkClient()
-        const instanceFilter: EC2.Filter[] = [
+    public getInstancesFilter(instanceIds: string[]): EC2.Filter[] {
+        return [
             {
                 Name: 'instance-id',
-                Values: [instanceId],
+                Values: instanceIds,
             },
         ]
+    }
+
+    /**
+     * Retrieve launch time of given EC2 instance.
+     * @param instanceId target EC2 instance ID
+     * @returns Date object for launch time associated with instance, or undefined if instance doesn't exists or doesn't have one.
+     */
+    public async getInstanceLaunchTime(instanceId: string): Promise<Date | undefined> {
+        const singleInstanceFilter = this.getInstancesFilter([instanceId])
+        try {
+            const instance = (await (await this.getInstances(singleInstanceFilter)).promise())[0]
+            return instance.LaunchTime!
+        } catch (err: unknown) {
+            return undefined
```

Comment:
silencing errors doesn't seem like the right pattern here. 

If an error occurs does getInstances() log it?

Reply:
It doesn't. This and the `IAMAttachDate` functions are both a little half-baked. I used them to debug the source of the error, but didn't up using them in the actual solution. I think its better if I just delete them.